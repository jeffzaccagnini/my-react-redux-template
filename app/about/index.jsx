export default () => <span>About page</span>;
