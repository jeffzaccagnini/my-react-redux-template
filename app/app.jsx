import { Link, Route, withRouter } from 'react-router-dom';
import { Portal } from 'react-redux-popup';
import Home from 'app/home';
import About from 'app/about';

export default props => (
    <div>
        <Link to="/">Home</Link>
        <Link to="/about">About</Link>
        <Route exact path="/" component={Home} />
        <Route exact path="/about" component={About} />
        <Portal
            modalTransitionName="modal"
            modalTransitionEnterTimeout={300}
            modalTransitionLeaveTimeout={300}
            popupTransitionName="popup"
            popupTransitionEnterTimeout={100}
            popupTransitionLeaveTimeout={100}
        />
    </div>
);

// export default withRouter(App);
