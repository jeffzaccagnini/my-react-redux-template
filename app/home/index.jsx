import { connect } from 'react-redux';
import { closePopup, openPopup } from 'react-redux-popup';
import PropTypes from 'prop-types';
import MyModal from 'app/home/modal';

export const Home = props => (
    <div>
        My Module
        <button onClick={() => props.openPopup('modal1')}>Open popup</button>
        <MyModal id="modal1" closePopup={props.closePopup}/>
    </div>
);

Home.propTypes = {
    closePopup: PropTypes.func.isRequired,
    openPopup: PropTypes.func.isRequired
};

export default connect(null, { closePopup, openPopup })(Home);
