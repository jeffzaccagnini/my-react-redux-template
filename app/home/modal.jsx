import PropTypes from 'prop-types';
import { Modal } from 'react-redux-popup';

const MyModal = props => (
    <div>
        <div>Some content</div>
        <button onClick={() => props.closePopup(props.id)}>Close</button>
    </div>
);

MyModal.propTypes = {
    closePopup: PropTypes.func.isRequired,
    id: PropTypes.string.isRequired
};

export default Modal(MyModal);
