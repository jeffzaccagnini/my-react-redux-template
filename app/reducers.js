import { combineReducers } from 'redux';
import { popupReducer } from 'react-redux-popup';
import { routerReducer } from 'react-router-redux';

export default combineReducers({
    popup: popupReducer,
    routing: routerReducer
});
