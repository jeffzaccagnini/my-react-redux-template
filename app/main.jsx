import { render } from 'react-dom';
import { createStore, applyMiddleware, compose } from 'redux';
import { Provider } from 'react-redux';
import createHistory from 'history/createBrowserHistory';
import { ConnectedRouter, routerMiddleware } from 'react-router-redux';
import thunk from 'redux-thunk';
import reducers from 'app/reducers';
import App from 'app/app';

// require style to generate one css file
require('../styles/main.scss');

function initializeDevToolsExtension() {
    return window.devToolsExtension({
        serializeAction: (key, value) => {
            if (typeof value === 'symbol') return String(value);
            return value;
        }
    });
}

const history = createHistory();

// middlewares
const reduxRouterMiddleware = routerMiddleware(history);
// eslint-disable-next-line no-nested-ternary
const devToolsExtension = process.env.NODE_ENV !== 'production'
    ? (window.devToolsExtension ? initializeDevToolsExtension() : (f => f))
    : (f => f);
const createStoreWithMiddleware = compose(
    applyMiddleware(reduxRouterMiddleware, thunk),
    devToolsExtension)(createStore);

// store
const store = createStoreWithMiddleware(reducers);

render(
    <Provider store={store}>
        <ConnectedRouter history={history}>
            <App />
        </ConnectedRouter>
    </Provider>,
    document.getElementById('main')
);
