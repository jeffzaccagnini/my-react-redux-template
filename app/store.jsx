import { browserHistory } from 'react-router';
import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import { routerMiddleware } from 'react-router-redux';
import reducers from 'rk/reducers';

function initializeDevToolsExtension() {
    return window.devToolsExtension({
        serializeAction: (key, value) => {
            if (typeof value === 'symbol') return String(value);
            return value;
        }
    });
}

const reduxRouterMiddleware = routerMiddleware(browserHistory);
const devToolsExtension = process.env.NODE_ENV !== 'production'
    ? (window.devToolsExtension ? initializeDevToolsExtension() : (f => f))
    : (f => f);
const createStoreWithMiddleware = compose(
    applyMiddleware(reduxRouterMiddleware, thunk),
    devToolsExtension)(createStore);
export default createStoreWithMiddleware(reducers);
