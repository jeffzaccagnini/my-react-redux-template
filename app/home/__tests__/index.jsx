import { shallow } from 'enzyme';
import { Home } from 'app/home';

describe('home/index', function() {
    it('should render the home component', function() {
        const handler = () => true;
        const tree = shallow(
            <Home closePopup={handler} openPopup={handler} />
        );
        expect(tree).toMatchSnapshot();
    });
});
