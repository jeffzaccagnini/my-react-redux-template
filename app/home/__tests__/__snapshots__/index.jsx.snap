// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home/index should render the home component 1`] = `
<div>
  My Module
  <button
    onClick={[Function]}
  >
    Open popup
  </button>
  <Connect(p)
    closePopup={[Function]}
    id="modal1"
  />
</div>
`;
